### PR TITLE
Stores - Laboratory Results AB#15701

### PR DESCRIPTION
**Pull request by BronzBierd**
#  Implements [AB#15701](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15701)

## Description

1. Split original laboratory store into general and COVID results
    - laboratory
    - covidLaboratory

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by BrianMaki — APPROVED** [review]

File: Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import { ActionType } from "@/constants/actionType";
+
+export const defaultCovid19TestResultState: Covid19TestResultState = {
+    data: [],
+    status: LoadStatus.NONE,
+    statusMessage: "",
+    error: undefined,
+};
+
+export const useLaboratoryStore = defineStore("laboratory", () => {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts:27]:
```suggestion
export const useCovid19TestResultStore = defineStore("covid19TestResult", () => {
```

File: Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts:1]: Recommend renaming to `covid19TestResult.ts` for consistency.

File: Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        SERVICE_IDENTIFIER.LaboratoryService
+    );
+    const datasetMapUtil = new DatasetMapUtils<
+        Covid19LaboratoryOrder[],
+        Covid19TestResultState
+    >(defaultCovid19TestResultState);
+
+    const errorStore = useErrorStore();
+
+    const labResultMap = ref(new Map<string, Covid19TestResultState>());
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts:39]: Rename to `covid19TestResultMap`.

File: Apps/WebClient/src/NewClientApp/src/stores/laboratory.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+export const defaultLabResultState: LabResultState = {
+    data: [],
+    status: LoadStatus.NONE,
+    statusMessage: "",
+    error: undefined,
+    queued: false,
+};
+
+export const useLaboratoryStore = defineStore("laboratory", () => {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/laboratory.ts:25]:
```suggestion
export const useLabResultStore = defineStore("labResult", () => {
```

File: Apps/WebClient/src/NewClientApp/src/stores/laboratory.ts
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/laboratory.ts:1]: Recommend renaming to `labResult.ts` for consistency.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts (hunk 1 above)
Comment (BronzBierd) [reply Apps/WebClient/src/NewClientApp/src/stores/covidLaboratory.ts:27]: this is a cloning error, I copied the whole file due to the similarities

**Pull request merged**